lib/vm.py

```diff
@@ -345,6 +347,14 @@ def destroy_vdi(self, vdi_uuid: str) -> None:
                 super().destroy_vdi(vdi_uuid)
                 break
 
+    def destroy_vdi_by_name(self, name: str) -> None:
+        for vdi in self.vdis:
+            if vdi.name() == name:
+                self.vdis.remove(vdi)
+                super().destroy_vdi(vdi.uuid)
```

Comment:
```suggestion
                vdi.destroy()
```
I think we should change this in `destroy_vdi` above, rewriting both to have the same logic could clean it up too.